fp/experiments.py

```diff
@@ -280,45 +280,70 @@ def run_single_exp(self, annotated_train_data, validation_data, test_data, scale
         results_file.to_csv(results_file_path, index=False)
 
 
+
     def filter_optimal_results(self):
         """Identifies the experiment(s) with the highest accuracy as optimal 
             result. Keeps the test metrics just for the experiment(s) with the
             optimal result.
         """
+
         results_dir_name = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../{}'.format(self.generate_file_path()))
         results_dir = os.listdir(Path(results_dir_name))
-        accuracies = dict()
-        max_accuracy = 0
 
-        # Fetching the accuracy from the row('val', 'None', 'accuracy') of all the experiment results
+        ##### SKYLINE FORMULA IMPLEMENTATION
+
+        privileged_metric_names = ['num_true_positives', 'num_false_positives', 'num_false_negatives',
+                                   'num_true_negatives', 'num_generalized_true_positives',
+                                   'num_generalized_false_positives', 'num_generalized_false_negatives',
+                                   'num_generalized_true_negatives', 'true_positive_rate', 'false_positive_rate',
+                                   'false_negative_rate', 'true_negative_rate', 'generalized_true_positive_rate',
+                                   'generalized_false_positive_rate', 'generalized_false_negative_rate',
+                                   'generalized_true_negative_rate', 'positive_predictive_value',
+                                   'false_discovery_rate', 'false_omission_rate', 'negative_predictive_value',
+                                   'accuracy', 'error_rate', 'num_pred_positives', 'num_pred_negatives',
+                                   'selection_rate']
+         
+        dictionary = {}
```

Comment:
please give this variable a different name so that a reader understand what is purpose is